### ISSUE TITLE
Question: use of Factor function

### ISSUE THREAD

**Issue by alecava58** [itinero/routing#17]
It is not clear for me if the vehicle profile Factor function is used only in route calculation or in creating contracted db too.
For example the BicycleBalanced profile class has an Factor function to avoid some ways, this function is used only when calculating the route?

Thanks in advance.

Regards

**Comment by xivk** [comment]
The factor function is the most important part of the profile and it gets used when creating a contracted db for that profile. It also get's used for other purposes like select start- and endlocations, or 'resolving'. By this I mean converting lat/lon to closest point on the network.

**Comment by alecava58** [comment]
So the factor function cannot contains variables that change at routing time.
Just to be clear. 
I'm trying to write a "mixed" factor function that has 2 variables (timeFactor and distanceFactor) and change those factors at runtime (sometime I want to calculate a route where the distance is more important, in this case I set distanceFactor = 0.7 and timeFactor = 0.3. Sometime the time is more important, in this case I set the distanceFactor = 0.3 and timeFactor = 0.7). 
If I understand well I must create a profile of each distanceFactor/timeFactor pair I need to use.
